torch/csrc/jit/passes/subgraph_rewrite.h

```diff
@@ -0,0 +1,98 @@
+/** \brief This file defines API for pattern-based subgraph rewrites.
+ *
+ * The API can be used for finding concrete patterns in the model and replacing
+ * the corresponding subgraphs with another subgraph. A special case of such
+ * rewrites is fusion, where the new subgraph consists of just a single node.
+ *
+ * There is a default set of most-common patterns that everyone could use, or
+ * alternatively an arbitrary pattern can be registered.
+ */
+#pragma once
+
+#include <torch/csrc/jit/ir.h>
+#include <torch/csrc/jit/script/module.h>
+
+#include <unordered_set>
+#include <vector>
+
+namespace torch {
+namespace jit {
+
+// Forward declarations.
+struct RewritePatternDescr;
+struct Match;
+
+/** \brief Run pattern-based subgraph rewrites on all methods in the module.
+ *
+ * This pass will go through all methods in the module and try to replace all
+ * recognized patterns (see SubgraphRewriter::RegisterDefaultPatterns for the
+ * list of these patterns).
+ */
+TORCH_API std::shared_ptr<script::Module> PatternBasedRewrite(
+    std::shared_ptr<script::Module> module);
+
+/** \brief A class implementing API for pattern-based subgraph rewrites.
```

Comment:
random style nit: I think `\brief` makes the comment less readable. For doxygen, you can configure so the first line in a javadoc-style comment is automatically taken as the brief description anyway.

So the comment could look like
```
/**
 * this is the brief description.
 * This is the rest
 */